crates/polars-core/src/series/any_value.rs

```diff
@@ -512,9 +512,18 @@ fn any_values_to_list(
     inner_type: &DataType,
     strict: bool,
 ) -> PolarsResult<ListChunked> {
-    let target_dtype = DataType::List(Box::new(inner_type.clone()));
+    let it = match inner_type {
+        // Structs don't support empty fields yet.
+        // We must ensure the data-types match what we do physical
+        #[cfg(feature = "dtype-struct")]
+        DataType::Struct(fields) if fields.is_empty() => {
```

Comment:
This is what we do physically. If field and physical don't match it doesn't uphold our invariants.